auslib/blobs/base.py

```diff
@@ -55,6 +55,41 @@ def createBlob(data):
     return blob_map[schema_version](**data)
 
 
+def merge_dicts(ancestor, left, right):
+    result = {}
+    for key in ancestor.keys() + left.keys() + right.keys():
+        key_types = set([type(ancestor.get(key)), type(left.get(key)), type(right.get(key))])
+        key_types.discard(type(None))
+        if len(key_types) > 1 and not key_types.issubset([str, unicode]):
+            raise ValueError("Cannot merge blobs: type mismatch for '{}'".format(key.encode('ascii', 'replace')))
+
+        if isinstance(ancestor.get(key), dict) or isinstance(left.get(key), dict) or isinstance(right.get(key), dict):
+            result[key] = merge_dicts(ancestor.get(key, {}), left.get(key, {}), right.get(key, {}))
+        elif isinstance(ancestor.get(key), list) or isinstance(left.get(key), list) or isinstance(right.get(key), list):
+            result[key] = ancestor.get(key, [])[:]
+            for side in (left, right):
+                for i in side.get(key, []):
+                    if i not in result[key] or not isinstance(i, type(result[key][result[key].index(i)])):
+                        result[key].append(i)
```

Comment:
it would be easier to test this if it was in a merge_lists function

Reply:
Yes....it would.